src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -93,17 +94,24 @@ private static boolean isValidPath(Path filePath) {
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
 
+        File storageFile = path.toFile();
+        if(!storageFile.exists()){
+            throw new StorageOperationException("Storage file not found!");           
+        }
+        
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
         try (final Writer fileWriter =
-                     new BufferedWriter(new FileWriter(path.toFile()))) {
+                     new BufferedWriter(new FileWriter(storageFile))) {
 
             final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (FileNotFoundException fnfe) {
+            throw new StorageOperationException("Storage file not file!");
```

Comment:
like what you said, FileNotFoundException doesn't work here but good job in trying ^^